tiledb/sm/query/result_tile.cc

```diff
@@ -910,14 +909,248 @@ void ResultTile::compute_results_count_sparse(
+  // find a `memcmp` on a much larger range.
+  const uint64_t zeroed_size = coords_num < 256 ? coords_num : 256;
+  for (uint64_t i = 0; i < coords_num; i += zeroed_size) {
+    const uint64_t partition_size =
+        (i < coords_num - zeroed_size) ? zeroed_size : coords_num - i;
+
+    // Check if all `r_bitmap` values are zero between `i` and
+    // `partition_size`.
+    if (has_previous_dim && r_bitmap[i] == 0 &&
+        memcmp(
+            &r_bitmap[i],
+            &r_bitmap[i + 1],
+            (partition_size - 1) * sizeof(uint8_t)) == 0)
+      continue;
+
+    {
+      // Here, we know that there is at least one `1` value within the
+      // `prev_dim_r_count` values within this partition. We must check
+      // each value for an intersection.
+      uint64_t c_offset = 0, c_size = 0;
+      for (uint64_t pos = i; pos < i + partition_size; ++pos) {
+        if (has_previous_dim && r_bitmap[pos] == 0)
+          continue;
+
+        c_offset = buff_off[pos];
+        c_size = (pos < coords_num - 1) ? buff_off[pos + 1] - c_offset :
+                                          buff_str_size - c_offset;
+        r_bitmap[pos] = str_coord_intersects(
+            c_offset, c_size, buff_str, range_start, range_end);
+      }
+    }
+  }
+}
+
+template <class T>
+void ResultTile::compute_results_bitmap_sparse(
+    const ResultTile* result_tile,
+    unsigned dim_idx,
+    const Range& range,
+    bool has_previous_dim,
+    std::vector<uint8_t>* result_bitmap,
+    const Layout& cell_order) {
+  // We don't use cell_order for this template type.
+  (void)cell_order;
+
+  // For easy reference.
+  auto coords_num = result_tile->cell_num();
+  auto r = (const T*)range.data();
+  auto stores_zipped_coords = result_tile->stores_zipped_coords();
+  auto dim_num = result_tile->domain()->dim_num();
+  auto& r_bitmap = (*result_bitmap);
+
+  // Variables for the loop over `coords_num`.
+  T c;
+  const T& r0 = r[0];
+  const T& r1 = r[1];
+
+  // Handle separate coordinate tiles
+  if (!stores_zipped_coords) {
+    const auto& coord_tile = std::get<0>(result_tile->coord_tile(dim_idx));
+    Buffer* const buffer = coord_tile.buffer();
+    const T* const coords = static_cast<const T*>(buffer->data());
+    {
+      if (has_previous_dim) {
+        for (uint64_t pos = 0; pos < coords_num; ++pos) {
+          c = coords[pos];
+          if (c >= r0 && c <= r1)
+            r_bitmap[pos] &= 1;
+          else
+            r_bitmap[pos] = 0;
+        }
+      } else {
+        for (uint64_t pos = 0; pos < coords_num; ++pos) {
+          c = coords[pos];
+          if (c >= r0 && c <= r1)
+            r_bitmap[pos] = 1;
+        }
+      }
+    }
+
+    return;
+  }
+
+  // Handle zipped coordinates tile
+  assert(stores_zipped_coords);
+  const auto& coords_tile = result_tile->zipped_coords_tile();
+  Buffer* const buffer = coords_tile.buffer();
+  const T* const coords = static_cast<const T*>(buffer->data());
+  {
+    if (has_previous_dim) {
```

Comment:
If you had initialized the initial bitmap to 1, then this would check would not be needed.